File: back-end/src/main/java/com/errasoft/friendfinder/model/Post.java
package com.errasoft.friendfinder.model;

import com.errasoft.friendfinder.model.security.Account;
import com.errasoft.friendfinder.utils.MediaType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "posts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Post extends BaseEntity {


    private String content;

    private String mediaUrl;

    @Enumerated(EnumType.STRING)
    private MediaType mediaType = MediaType.NONE;


    @Column(nullable = false)
    private boolean deleted = false;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private Account owner;

    private Long likesCount = 0L;
    private Long dislikesCount = 0L;
    private Long commentsCount = 0L;
}

File: front-end/src/model/post.ts
import {Account} from './account';
import {Comment} from './comment';

export class Post {
  id: number;
  content: string;
  mediaUrl: string;
  mediaType: 'IMAGE' | 'VIDEO' | 'NONE';
  owner: Account;
  likesCount: number;
  commentsCount: number;
  dislikesCount: number;
  createdDate: string;
  comments: Comment[] = [];
  commentPage = 1;
  showMore = true;
  showDropdown = false;
  newComment?: string;
  showDeleteConfirm?: boolean;
}
